app/code/Magento/CatalogImportExport/Model/Import/Uploader.php

```diff
@@ -198,7 +198,7 @@ public function move($fileName, $renameFileOff = false)
             );
         }
 
-        $filePath = $this->_directory->getRelativePath($filePath . $fileName);
+        $filePath = strpos($filePath, $fileName) !== false ? $filePath:$this->_directory->getRelativePath($filePath . $fileName);
```

Comment:
Please report issue first so that it can be verified for 2.2 and 2.3, then 2.3 should be fixed first if there is a problem.

Reply:
I think have issue for it.
https://github.com/magento/magento2/issues/2666
But It's close. It's only happen with external image when you import a product.
I found the problem come from here:
If the image is external it will be run this condition

At line 194: app/code/Magento/CatalogImportExport/Model/Import/Uploader.php 
They merge `$fileName` to the path.

`        if (preg_match('/\bhttps?:\/\//i', $fileName, $matches)) {
            $url = str_replace($matches[0], '', $fileName);
            $driver = $matches[0] === $this->httpScheme ? DriverPool::HTTP : DriverPool::HTTPS;
            $read = $this->_readFactory->create($url, $driver);
            //only use filename (for URI with query parameters)
            $parsedUrlPath = parse_url($url, PHP_URL_PATH);
            if ($parsedUrlPath) {
                $urlPathValues = explode('/', $parsedUrlPath);
                if (!empty($urlPathValues)) {
                    $fileName = end($urlPathValues);
                }
            }
            $fileExtension = pathinfo($fileName, PATHINFO_EXTENSION);
            if ($fileExtension && !$this->checkAllowedExtension($fileExtension)) {
                throw new \Magento\Framework\Exception\LocalizedException(__('Disallowed file type.'));
            }
            $fileName = preg_replace('/[^a-z0-9\._-]+/i', '', $fileName);
            $relativePath = $this->_directory->getRelativePath($filePath . $fileName);
            $this->_directory->writeFile(
                $relativePath,
                $read->readAll()
            );
        }`

At line 201: They merged $fileName to the path again 
`
$relativePath = $this->_directory->getRelativePath($filePath . $fileName);
`
now the name is duplicated. and it became `path/filenamefilename`. This file is not exist on system so the import is return `Imported resource (image) could not be downloaded from external resource due to timeout or access permissions`